controllers/drplacementcontrol_controller.go

```diff
@@ -2422,3 +2447,173 @@ func (r *DRPlacementControlReconciler) determineDRPCState(
 
 	return AllowFailover, msg, nil
 }
+
+// ensureVRGsManagedByDRPC ensures that VRGs reported by ManagedClusterView are managed by the current instance of
+// DRPC. This is done using the DRPC UID annotation on the viewed VRG matching the current DRPC UID and if not
+// creating or updating the exisiting ManifestWork for the VRG.
+// Returns a bool indicating true if VRGs are managed by the current DRPC resource
+func ensureVRGsManagedByDRPC(
+	log logr.Logger,
+	mwu rmnutil.MWUtil,
+	vrgs map[string]*rmn.VolumeReplicationGroup,
+	drpc *rmn.DRPlacementControl,
+	vrgNamespace string,
+) bool {
+	ensured := true
+
+	for cluster, viewVRG := range vrgs {
+		if viewVRG == nil {
```

Comment:
(nit) should never be nil. So maybe just eliminate this check.

Reply:
done